File: week4/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k % nums.length;
        if (k == 0) {
            return;
        }
        Queue<Integer> q = new LinkedList<Integer>();
        for (int i = nums.length - k; i < nums.length; i++) {
            q.add(nums[i]);
        }
        for (int i = 0; i < nums.length - k; i++) {
            q.add(nums[i]);
        }
        int i = 0;
        while (i < nums.length) {
            nums[i] = q.remove();
            i++;
        }
    }
}

File: summer2021/unit4/rotate-array.py
from collections import deque
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        queue = deque()
        for i in nums:
            queue.append(i)
        i = k % len(nums)
        while len(queue) > 0:
            nums[i] = queue.popleft()
            i = (i + 1) % len(nums)
        